de.nagstamon.nagstamon.yml

```diff
@@ -0,0 +1,115 @@
+  - --filesystem=xdg-run/keyring
+  - --filesystem=home
+cleanup-commands:
+  - /app/cleanup-BaseApp.sh
+modules:
+  - name: libsecret
+    buildsystem: meson
+    config-opts:
+      - -Dgtk_doc=false
+      - -Dintrospection=false
+      - -Dmanpage=false
+      - -Dvapi=false
+    cleanup:
+      - /bin
+      - /include
+      - /lib/pkgconfig
+      - /share/man
+    sources:
+      - type: archive
+        url: https://download.gnome.org/sources/libsecret/0.21/libsecret-0.21.4.tar.xz
+        sha256: 163d08d783be6d4ab9a979ceb5a4fecbc1d9660d3c34168c581301cd53912b20
+
+  - name: krb5
+    subdir: src
+    config-opts:
+      - --localstatedir=/var/lib
+      - --sbindir=${FLATPAK_DEST}/bin
+      - --disable-static
+      - --disable-rpath
+    cleanup:
+      - /bin
+      - /share/et
+      - /share/examples
+      - /share/man
+    sources:
+      - type: archive
+        url: https://github.com/krb5/krb5/archive/refs/tags/krb5-1.21.2-final.tar.gz
+        sha256: 5827fa39ead0e7f7b8a158799eb5d50b194129e81b378892caedcafe50c5c195
+      - type: shell
+        dest: src
+        commands:
+          - autoreconf -si
+
+  - name: dbus-run-session
+    buildsystem: autotools
+    cleanup: ['*']
+    sources:
+      - type: archive
+        url: https://gitlab.freedesktop.org/dbus/dbus/-/archive/dbus-1.14.10/dbus-dbus-1.14.10.tar.gz
+        sha256: 5fedcd6c17fbc62646af224e1aab11a334c463e7e7d89163a670ed24e7ab241c
+        x-checker-data:
+          type: anitya
+          project-id: 5356
+          url-template: https://gitlab.freedesktop.org/dbus/dbus/-/archive/dbus-$version/dbus-dbus-$version.tar.g00z
+
+  - name: dbus-python
+    buildsystem: autotools
+    sources:
+      - type: archive
+        url: https://dbus.freedesktop.org/releases/dbus-python/dbus-python-1.3.2.tar.gz
+        sha256: ad67819308618b5069537be237f8e68ca1c7fcc95ee4a121fe6845b1418248f8
+        x-checker-data:
+          type: anitya
+          project-id: 402
+          url-template: https://dbus.freedesktop.org/releases/dbus-python/dbus-python-$version.tar.gz
+
+  # external python modules
+  - python-modules.yml
+
+  - name: nagstamon
+    buildsystem: simple
+    build-commands:
+      - python3 setup.py bdist --format tar
+      - tar -xvf dist/nagstamon*.tar --strip 2 -C /app
+      - install -D nagstamon.sh /app/bin/nagstamon
+      - mkdir -p /app/share/icons/hicolor/scalable/apps/
+      - patch -i nagstamon_svg_size.patch /app/share/pixmaps/nagstamon.svg
+      - install -Dm644 /app/share/pixmaps/nagstamon.svg /app/share/icons/hicolor/scalable/apps/
+      - install -Dm644 nagstamon.appdata.xml -t /app/share/metainfo
+    sources:
+      - type: archive
+        url: https://github.com/HenriWahl/Nagstamon/archive/refs/tags/v3.14.0.tar.gz
+        sha256: f51c50feb7efa328d252c638b6602405d550a99622dd7e8f07341814879e9f30
+      - type: file
+        url: https://raw.githubusercontent.com/HenriWahl/Nagstamon/master/Nagstamon/resources/nagstamon.appdata.xml
```

Comment:
this should reference a specific commit instead of `master`. Otherwise the build will break as soon as it is changed.

Reply:
Sounds sensible. So i I want to use it with the stable branch 3.14.0 this appdata.xml should reside exactly in this branch, right?

Is there a way to use variables in the manifest like ${version} to avoid to write the same version in various places/URLs?

Reply:
a commit is a commit. the branch it is on is irrelevant.

this should no longer be needed in the next release based off `master`.

Reply:
So if I use the URL https://raw.githubusercontent.com/HenriWahl/Nagstamon/5d994ea42ad86bbbdc61669b8a45fecaa0183f14/Nagstamon/resources/nagstamon.appdata.xml it will fulfill you requirement as being pinned to a certain version?

Reply:
@hfiguiere works so far. But I am still wondering about the version `master`:

```
flatpak --user install local-repo de.nagstamon.nagstamon                                           de.nagstamon.nagstamon  
Suchen nach Übereinstimmungen …

de.nagstamon.nagstamon Berechtigungen:
    ipc      network     pulseaudio     x11     dri     file access [1]     dbus access [2]

    [1] home, xdg-config/kdeglobals:ro, xdg-run/keyring
    [2] com.canonical.AppMenu.Registrar, org.freedesktop.Notifications, org.freedesktop.secrets, org.kde.KGlobalSettings, org.kde.kconfig.notify


        KENNUNG                                  Zweig            Op           Gegenstelle           Herunterladen
 1. [✓] de.nagstamon.nagstamon.Locale            master           i            local-repo            1,0 KB / 61,1 MB
 2. [✓] de.nagstamon.nagstamon                   master           i            local-repo            1,0 KB / 110,6 MB
```

Is this to be expected locally or shouldn't there appear the version from https://github.com/HenriWahl/Nagstamon/blob/5d994ea42ad86bbbdc61669b8a45fecaa0183f14/Nagstamon/resources/nagstamon.appdata.xml#L12 ?

Reply:
this wasn't done.

Reply:
Sorry? I found out that version number is the one in the local devel branch but as @bbhtt wrote ago the result of `flatpak info de.nagstamon.nagstamon` looks OK:

```
Nagstamon - The status monitor for the desktop

         Kennung: de.nagstamon.nagstamon
             Ref: app/de.nagstamon.nagstamon/x86_64/master
     Architektur: x86_64
           Zweig: master
         Version: 3.14.0
          Lizenz: GPL-2.0
        Ursprung: local-repo
        Sammlung: 
    Installation: user
     Installiert: 281,4 MB
Laufzeitumgebung: org.kde.Platform/x86_64/6.6
             Sdk: org.kde.Sdk/x86_64/6.6

          Commit: ebfb62af3dd8a9448395622eabe78d33c30c29820b671dcbf357f59005231892
         Betreff: Export de.nagstamon.nagstamon
           Datum: 2024-04-21 15:01:01 +0000
```

Reply:
Maybe because we are talking about git revision and branches. Nothing else.

Reply:
OK so it is OK.